consensus/core/src/core.rs

```diff
@@ -87,6 +87,9 @@ pub(crate) struct Core {
     /// This is currently being used to avoid equivocations during a node recovering from amnesia. When value is None it means that
     /// the last block sync mechanism is enabled, but it hasn't been initialised yet.
     last_known_proposed_round: Option<Round>,
+    /// When true it will block the block proposal until it has successfully completed the process
```

Comment:
It seems this never updates. Is the reason we need this field in addition to `last_known_proposed_round`, because of unit testing? Maybe we can default the timeout to 0 for cfg(test) as well.

Reply:
Not for testing but for checking if amnesia recovery is enabled when trying to call `set_last_known_proposed_round`. I've refactored, indeed we don't _really_ needed as we can infer from whether the `last_known_proposed_round` is already set or not.